### PR TITLE
[libc++] Don't commit libcxx.imp

### PR DESCRIPTION
**Pull request by ldionne**
We can instead generate it on-the-fly when we install the headers. This reduces the amount of boilerplate we have to re-generate whenever we add, remove or relocate header files.

Fixes #88529

### CODE REVIEW

File: libcxx/utils/CMakeLists.txt [hunk 1]
```diff
@@ -71,5 +65,4 @@ add_custom_target(libcxx-generate-files
             libcxx-generate-escaped-output-table
             libcxx-generate-width-estimation-table
 			libcxx-indic-conjunct-break-table
-            libcxx-generate-iwyu-mapping
```
Comment (mordante) [inline libcxx/utils/CMakeLists.txt:74]: Can you also fix the tab I sneaked in on the line above?

File: libcxx/include/CMakeLists.txt [hunk 2]
```diff
@@ -1036,6 +1035,15 @@ foreach(f ${files})
...
 endforeach()
 
+# Generate the IWYU mapping. This depends on all header files but it's also considered as an
+# "include" for dependency tracking.
+add_custom_command(OUTPUT "${LIBCXX_GENERATED_INCLUDE_DIR}/libcxx.imp"
+  COMMAND "${Python3_EXECUTABLE}" "${LIBCXX_SOURCE_DIR}/utils/generate_iwyu_mapping.py" "-o" "${LIBCXX_GENERATED_INCLUDE_DIR}/libcxx.imp"
+  DEPENDS ${_all_includes}
+  COMMENT "Generate the mapping file for include-what-you-use"
+)
+list(APPEND _all_includes "${LIBCXX_GENERATED_INCLUDE_DIR}/libcxx.imp")
```
Comment (mordante) [inline libcxx/include/CMakeLists.txt:1045]: It looks like this breaks the CI.

**Review by mordante — CHANGES_REQUESTED** [review]
Thanks for the cleanup. I like to have a short look after it passes the CI.

**Review by mordante — APPROVED** [review]
I see some CI's passing, so I'm happy. LGTM!

**Comment by nico** [comment]
Hello, is there any way this could be reconsidered? We have a bot that auto-updates libc++ in Chromium, and that bot only watches for revisions and produces CLs without having any local checkout state. This change here makes it impossible for that bot to continue working in its current state.

("Can't you just change the bot?" Not easily. It's a general dependency-updating system that we use for many dependencies, and not having to have local checkouts is a design decision that the team setting up that system made.)

If it's a big pain to keep upstream, we'll find some way I suppose, but I figured asking first is easy, and if it isn't painful to keep this upstream…

**Comment by ldionne** [comment]
I am not sure I understand what's problematic about this specific patch w.r.t. that bot, can you explain?

We generate several files in the installation directory (e.g. `__config_site` and `__assertion_handler`) already, how is this one different?

**Comment by nico** [comment]
We don't use any of the other generated files either. We have fixed `__config_site` and `__assertion_handler` files in our tree. They never (or rarely) change, and if they do, something tends to break loudly if we don't update our version. This file here changes more often, and the error case is that IWYU silently adds includes of internal headers.

**Comment by ldionne** [comment]
So, are you basically asking whether we would be comfortable with not generating files into the install directory without having them checked in somewhere, in the general case?

Sorry, I'm genuinely trying to understand what your request entails before I can really form an opinion.

**Comment by nico** [comment]
> So, are you basically asking whether we would be comfortable with not generating files into the install directory without having them checked in somewhere, in the general case?

Yes, except I'm not asking about the general case, but about this specific case :)

**Pull request merged**